### PR TITLE
Add buildDiscarder and timestamp option to Jenkins

### PR DESCRIPTION
**Pull request by BrennanGit**
This PR introduces a method for automatically deleting Jenkins builds after a certain time/number. This is needed so that Jenkins disk does not forever grow.

Currently this repo is using 133GB of space on the Jenkins master and now archiving 2.3GB every build!
Reducing this would not only stop me needing to respec the jenkins master but should also make it run faster.

Please understand that **THIS IS A DESTRUCTIVE CHANGE** and once merged to develop will permanently delete old builds so please come forward with any concerns over the settings chosen if you have any.

We will use the [buildDiscarder Jenkinsfile option](https://stackoverflow.com/questions/39542485/how-to-write-pipeline-to-discard-old-builds/44155346#44155346) to configure this on a repo-by-repo basis.

This process has been tested using [this repo](https://github.com/xmos/jenkins_test) and [this jenkins job](http://srv-bri-jcim0:8080/job/XMOS/job/jenkins_test/).

Discussions about adding it to a previous repo are here: https://github.com/xmos/lib_aec/pull/448

------------------------

[Here is a link to the confluence page](https://xmosjira.atlassian.net/wiki/spaces/wiki/pages/3396534277/Build+Discarding) going into more detail about justification and usage.

The short version is we propose to delete any build older than 100 on the develop branch of this repo.

Please comment if you think this is unsuitable.

------------------------

PS I have also opted to use this opportunity to add the timestamps option to jobs that do not already employ it. This makes it easier to diagnose any future Jenkins issues.

### CODE REVIEW

**Review by keithm-xmos — APPROVED** [review]

**Comment by BrennanGit** [comment]
The number here is the last 100 builds (matching the other times this change has been added) but in this case, that is unlikely to reduce the current space used by much. 
Can I then ask if we can set it to a lower number, say 50?
Also I might suggest wrapping the archiveArtifacts of the wavs in some sort of when block so they are only archived when necessary.

**Comment by mbanth** [comment]
> Can I then ask if we can set it to a lower number, say 50?

50 runs is fine with me.

**Review by mbanth — APPROVED** [review]

**Comment by BrennanGit** [comment]
Are we ok to merge this once the build has passed? @shuchitak @mbanth

**Comment by mbanth** [comment]
> Are we ok to merge this once the build has passed? @shuchitak @mbanth

Merging after a successful build is fine with me.

**Comment by shuchitak** [comment]
> Are we ok to merge this once the build has passed? @shuchitak @mbanth

Yes, I'm okay with merging this.
